File: Data Structures & Algorithms/duplicate-integer/submission-5.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::set<int> numSet;

        for(int num : nums) {
            if(numSet.count(num)) {
                return true;
            }
            numSet.insert(num);
        }

        return false;
    }
};

File: Data Structures & Algorithms/majority-element/submission-5.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums: number[]): number {
        let candidate: number = 0;
        let count: number = 0;

        for(let i: number = 0; i < nums.length; ++i) {
            if(count == 0) {
                candidate = nums[i];
            }
            if(candidate == nums[i]) {
                count++;
            } else {
                count--;
            }
        }
        return candidate;
    }
}
